packages/eslint-plugin/src/util/explicitReturnTypeUtils.ts

```diff
@@ -256,11 +253,20 @@ function isValidFunctionExpressionReturnType(
   }
 
   // https://github.com/typescript-eslint/typescript-eslint/issues/653
-  return (
-    options.allowDirectConstAssertionInArrowFunctions === true &&
-    node.type === AST_NODE_TYPES.ArrowFunctionExpression &&
-    returnsConstAssertionDirectly(node)
-  );
+  if (
+    !options.allowDirectConstAssertionInArrowFunctions ||
+    node.type !== AST_NODE_TYPES.ArrowFunctionExpression
+  ) {
+    return false;
+  }
+
+  const { body } = node;
+
+  if (body.type === AST_NODE_TYPES.TSSatisfiesExpression) {
+    return isConstAssertion(body.expression);
+  }
+
+  return isConstAssertion(body);
```

Comment:
[Style] This is super nitpicky, but I found collapsing down to one statement to be a bit more readable:

```suggestion
  return isConstAssertion(
    body.type === AST_NODE_TYPES.TSSatisfiesExpression ? body.expression : body,
  );
```

No worries at all if you hate this 😄

Reply:
Not my cup of tea (🍵) but is is more concise 👍